### PR TITLE
Mock defaults for PTYTextViewTests

### PR DESCRIPTION
**Pull request by jgavris**
NB: Make sure to open `iTerm2.xcworkspace` from now on. Most of the big scary diff is just checking in the pods (some `.gitignore` the `Pods/` directory, I consider it part of the application source).

### CODE REVIEW

**Comment by gnachman** [comment]
Any idea why there would be two identically named targets (the first and last ones in the screenshot below)? One builds and one doesn't. 
screen shot 2016-01-21 at 8 04 16 pm

**Comment by gnachman** [comment]
I tried to run the tests by following these steps:

1. Select iTerm2Tests as the active scheme
2. Click the button next to PTYTextViewTests

It fails to build the target iTerm2XCTests with this error:
````
/Users/gnachman/iTerm2/iTerm2XCTests/PTYTextViewTest.m:16:9: fatal error: 'OCHamcrest/OCHamcrest.h' file not found
#import <OCHamcrest/OCHamcrest.h>
        ^
1 error generated.
````

File: iTerm2XCTests/SelectorSwizzler.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+@import Foundation;
+
+/**
+ * A helper class for swizzling a selector for the scope of a block.
+ */
+@interface SelectorSwizzler : NSObject
```
Comment (gnachman) [inline iTerm2XCTests/SelectorSwizzler.h]: New classes should be named like `iTermSelectorSwizzler` (there are a some holdouts that I haven't renamed yet, but I'm moving toward this scheme.)

File: iTerm2XCTests/SelectorSwizzler.m [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#import "SelectorSwizzler.h"
+#import <objc/runtime.h>
+
+@implementation SelectorSwizzler
+
++ (void)swizzleSelector:(SEL)selector
+              fromClass:(Class)fromClass
+              withBlock:(id)fakeSelectorBlock
+               forBlock:(dispatch_block_t)block
+{
```
Comment (gnachman) [inline iTerm2XCTests/SelectorSwizzler.m]: Move this brace up to the previous line.

**Comment by jgavris** [comment]
Did another pass, the `Pods.xcodeproj` was being `.gitignored`. That would explain the initial failure on Travis. This is what my schemes look like now with a clean working tree.

screen shot 2016-01-21 at 11 29 04 pm

**Comment by gnachman** [comment]
I still can't run tests. With either the iTerm2 or the iTerm2Tests scheme selected I get this:

````
CompileC build/iTerm2.build/Development/iTerm2XCTests.build/Objects-normal/x86_64/PTYTextViewTest.o iTerm2XCTests/PTYTextViewTest.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/gnachman/git/fork/test-coverage
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu99 -fmodules -gmodules -fmodules-cache-path=/Users/gnachman/Library/Developer/Xcode/DerivedData/ModuleCache -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/gnachman/Library/Developer/Xcode/DerivedData/ModuleCache/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-shorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wno-deprecated-implementations -DDEBUG=1 -DCOCOAPODS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -mmacosx-version-min=10.10 -g -Wno-sign-conversion -iquote /Users/gnachman/git/fork/test-coverage/build/iTerm2.build/Development/iTerm2XCTests.build/iTerm2XCTests-generated-files.hmap -I/Users/gnachman/git/fork/test-coverage/build/iTerm2.build/Development/iTerm2XCTests.build/iTerm2XCTests-own-target-headers.hmap -I/Users/gnachman/git/fork/test-coverage/build/iTerm2.build/Development/iTerm2XCTests.build/iTerm2XCTests-all-non-framework-target-headers.hmap -ivfsoverlay /Users/gnachman/git/fork/test-coverage/build/iTerm2.build/all-product-headers.yaml -iquote /Users/gnachman/git/fork/test-coverage/build/iTerm2.build/Development/iTerm2XCTests.build/iTerm2XCTests-project-headers.hmap -I/Users/gnachman/git/fork/test-coverage/build/Development/include -I/Users/gnachman/git/fork/test-coverage/build/iTerm2.build/Development/iTerm2XCTests.build/DerivedSources/x86_64 -I/Users/gnachman/git/fork/test-coverage/build/iTerm2.build/Development/iTerm2XCTests.build/DerivedSources -Wall -F/Users/gnachman/git/fork/test-coverage/build/Development -F/Applications/Xcode.app/Contents/Developer/Library/Frameworks -F/Users/gnachman/git/fork/test-coverage -F/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks -iquote /Users/gnachman/git/fork/test-coverage/build/Development/OCHamcrest.framework/Headers -iquote /Users/gnachman/git/fork/test-coverage/build/Development/OCMockito.framework/Headers -MMD -MT dependencies -MF /Users/gnachman/git/fork/test-coverage/build/iTerm2.build/Development/iTerm2XCTests.build/Objects-normal/x86_64/PTYTextViewTest.d --serialize-diagnostics /Users/gnachman/git/fork/test-coverage/build/iTerm2.build/Development/iTerm2XCTests.build/Objects-normal/x86_64/PTYTextViewTest.dia -c /Users/gnachman/git/fork/test-coverage/iTerm2XCTests/PTYTextViewTest.m -o /Users/gnachman/git/fork/test-coverage/build/iTerm2.build/Development/iTerm2XCTests.build/Objects-normal/x86_64/PTYTextViewTest.o

/Users/gnachman/git/fork/test-coverage/iTerm2XCTests/PTYTextViewTest.m:16:9: fatal error: 'OCHamcrest/OCHamcrest.h' file not found
#import <OCHamcrest/OCHamcrest.h>
        ^
1 error generated.
````

**Comment by jgavris** [comment]
That's strange, what command are you using? Do you have the `.xcworkspace` file open (you will see a `Pods` project as well in the workspace). I can cmd + U the iTerm2 scheme and also run `./ci/script.sh` (which is what travis runs, and passed the build).

**Pull request merged**